homeassistant/components/flo/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for flo integration."""
+import logging
+
+from aioflo import async_get_api
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"username": str, "password": str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    session = async_get_clientsession(hass)
+    try:
+        api = await async_get_api(
+            data[CONF_USERNAME], data[CONF_PASSWORD], session=session
+        )
+    except Exception:  # pylint: disable=broad-except
```

Comment:
Can we catch more specific errors?

Reply:
updated

Reply:
Should we catch it though? Validate Input is already called from inside a try block that catches generic exception, and is able to handle it correctly ("unknown error").

Reply:
Also same comment about being able to detect invalid auth.